packages/main/src/ResponsivePopover.ts

```diff
@@ -1,8 +1,17 @@
+import type { ComponentStylesData } from "@ui5/webcomponents-base/dist/types.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
 import { isPhone } from "@ui5/webcomponents-base/dist/Device.js";
+import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import type { I18nText } from "@ui5/webcomponents-base/dist/i18nBundle.js";
 import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
 import { getNextZIndex } from "@ui5/webcomponents-base/dist/util/PopupUtils.js";
+
+// @ts-ignore
 import { RESPONSIVE_POPOVER_CLOSE_DIALOG_BUTTON } from "./generated/i18n/i18n-defaults.js";
+
 import ResponsivePopoverTemplate from "./generated/templates/ResponsivePopoverTemplate.lit.js";
+import type { BeforeCloseEventDetail } from "./Popup.js";
```

Comment:
latest convention of naming this kind of types is PopupBeforeCloseEventDetail. For consistency maybe we have to introduce new type ResponsivePopoverBeforeCloseEventDetail = PopupBeforeCloseEventDetail. @ilhan007 what do u think?

Reply:
I agree
- BeforeCloseEventDetail should be renamed to PopupBeforeCloseEventDetail
- For the second we don't have a hard convention, but I think 
type ResponsivePopoverBeforeCloseEventDetail = PopupBeforeCloseEventDetail is fine

Reply:
> BeforeCloseEventDetail should be renamed to PopupBeforeCloseEventDetail
Renamed this.
Though, I don't think it's necessary to introduce ResponsivePopoverBeforeCloseEventDetail because it doesn't get overriden, the regular "before-close" is propagated upwards here.

Reply:
If somehow these event details are not the same (f.e we decide to add parameter only for ResponsivePopover), developers will continue to use PopupBeforeCloseEventDetail instead of the correct one and they may not be able to access that new parameter.